app/controllers/api/v2/environments_controller.rb

```diff
@@ -61,6 +61,13 @@ def destroy
       def allowed_nested_id
         %w(puppetclass_id location_id organization_id)
       end
+
+      def find_resource
+        not_found and return if params[:id].blank?
+        resource = resource_scope.find_by_name(params[:id]) rescue nil
```

Comment:
to be compatible with rails 4 use where(:name => params[:id]).first, and no need to rescue